genie_registry/maf.py

```diff
@@ -277,6 +277,63 @@ def _validate(self, mutationDF):
 
         return total_error.getvalue(), warning.getvalue()
 
+    def _cross_validate(self, mutationDF: pd.DataFrame) -> tuple:
+        """This function cross-validates the mutation file to make sure it
+        adheres to the mutation SOP.
+
+        Args:
+            mutationDF (pd.DataFrame): mutation dataframe
+
+        Returns:
+            Text with all the errors in the mutation file
+        """
+        errors = ""
+        warnings = ""
+        maf_filename = f"data_mutations_extended_{self.center}.txt"
+
+        # This section can be removed once we remove the list of lists
+        clinical_files = validate.parse_file_info_in_nested_list(
+            nested_list=self.ancillary_files, search_str="data_clinical_supp"
+        )
+        clinical_file_names = clinical_files["file_info"]["name"]
+        clinical_file_paths = clinical_files["file_info"]["path"]
+
+        if clinical_files["files"]:
+            try:
+                clinical_sample_df = process_functions.get_clinical_dataframe(
+                    filePathList=clinical_file_paths
+                )
+            except Exception as e:
+                errors = f"The clinical file(s) cannot be read. No cross-validation will be done. Original error: {str(e)}"
+                warnings = ""
+
+            if not errors:
+                if process_functions.checkColExist(
+                    mutationDF, "SAMPLE_ID"
+                ) and process_functions.checkColExist(
+                    clinical_sample_df, "TUMOR_SAMPLE_BARCODE"
+                ):
+                    errors, warnings = validate.check_values_between_two_df(
+                        df1=mutationDF,
+                        df1_filename=maf_filename,
+                        df1_id_to_check="SAMPLE_ID",
+                        df2=clinical_sample_df,
+                        df2_filename=clinical_file_names,
+                        df2_id_to_check="TUMOR_SAMPLE_BARCODE",
+                    )
+                else:
+                    errors = ""
+                    warnings = (
+                        "SAMPLE_ID doesn't exist in the maf or TUMOR_SAMPLE_BARCODE doesn't exist clinical file(s)."
```

Comment:
You don't need to do the sample Id check in the maf because that is done in the row based validation step.

If initial variation fails, no cross file validation is done.